### DISCUSSION TITLE
Array variable

### DISCUSSION THREAD

**Discussion by mremadevi** [elsa-workflows/elsa-core#5333] [Q&A]
How to append item to string array variable using csharp script.

**Comment by glime-ai** [comment]
To append an item to a string array variable using C# script in Elsa Workflow, you would typically use the `List<T>.Add` method if you are dealing with a `List<string>`. However, if you are indeed working with an array (`string[]`), you would need to create a new array that includes the new item since arrays have a fixed size once created.

Here's an example of how you might do this in a C# script within an Elsa Workflow:

```csharp
// Assuming 'myStringArray' is a variable of type string[] already defined in the workflow
var myStringArray = Variables.Get<string[]>("myStringArray");
var itemToAdd = "new item"; // The new item you want to append

// Create a new array with an additional slot for the new item
var newArray = new string[myStringArray.Length + 1];
myStringArray.CopyTo(newArray, 0); // Copy existing items to the new array
newArray[newArray.Length - 1] = itemToAdd; // Set the last item to the new item

// Update the workflow variable with the new array
Variables.Set("myStringArray", newArray);
```

Please note that in the context of Elsa Workflow, you would use the `Variables` object to get and set the workflow variables. The above script assumes that you have a variable named `myStringArray` that is an array of strings.

If you are using a `List<string>` instead, the code would be simpler:

```csharp
// Assuming 'myStringList' is a variable of type List<string> already defined in the workflow
var myStringList = Variables.Get<List<string>>("myStringList");
var itemToAdd = "new item"; // The new item you want to append

// Add the new item to the list
myStringList.Add(itemToAdd);

// Update the workflow variable with the modified list
Variables.Set("myStringList", myStringList);
```

In this case, since `List<T>` is a dynamic collection, you can directly add new items to it without needing to create a new instance.

Remember to replace `"myStringArray"` or `"myStringList"` with the actual name of your variable, and `"new item"` with the actual string you want to append.

If you need further clarification or have additional context, please let me know!

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
